src/github.com/getlantern/flashlight/ui/service.go

```diff
@@ -0,0 +1,149 @@
+package ui
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+)
+
+type helloFnType func(func(interface{}) error) error
+
+type Service struct {
+	Type       string
+	In         <-chan interface{}
+	Out        chan<- interface{}
+	in         chan interface{}
+	out        chan interface{}
+	newMessage func() interface{}
+	helloFn    helloFnType
+}
+
+var (
+	mu               sync.RWMutex
+	defaultUIChannel *UIChannel
+
+	services = make(map[string]*Service)
+)
+
+func (s *Service) write() {
+	// Watch for new messages and send them to the combined output.
+	for msg := range s.out {
+		b, err := newEnvelope(s.Type, msg)
+		if err != nil {
+			log.Error(err.Error())
```

Comment:
Would not it be better to use continue instead of sending a null value to defaultUIChannel.Out? No need to send empty messages to clients.